cilium/CFP-27752-operator-manages-cilium-identities.md

```diff
@@ -0,0 +1,220 @@
+
+Simultaneous CID management is only expected to happen as a transitional state during upgrade from Agents managing CIDs to Operator managing CIDs.
+
+
+## CID deduplication
+
+Operator will not create duplicate CIDs, but it will allow for their existence. In case other sources, like Agents, are making duplicate CIDs, Operator will be compatible with tracking multiple CIDs. This is needed to make Operator compatible with Agents managing CIDs at the same time.
+
+Completely deduplicating CIDs could be done with additional changes in Agent. One option is to have Agents always pick CIDs with the lowest integer value in the name, that matches the label set. This way all Agents would always pick the same CID for the same label set. However, this is only a suggestion, and a proper solution will be covered separately, outside this CFP.
+
+
+## KVStore compatibility
+
+**Important note: kvstore compatibility is a hard requirement for releasing the feature "operator manages CIDs".**
+
+[KVStore](https://docs.cilium.io/en/stable/kvstore/) is a key-value store that is mostly used for sharing Cilium mappings between IPs, identities (IPCache) and labels, across the cluster ([what's inside Cilium’s KVStore](http://arthurchiao.art/blog/whats-inside-cilium-etcd/)).
+
+Operator managing CIDs is intended to work with both `identity-allocation-mode=kvstore` and `identity-allocation-mode=crd`. CRD being the default, the initial implementation will use the CRD allocation mode, and later expanded to other allocation modes.
+
+All features that rely on using kvstore will not work until CID controller is expanded to work with kvstore allocation mode. The work to enable kvstore compatibility is estimated to of medium size and it will be tracked in a separate GitHub issue (https://github.com/cilium/cilium/issues/34865).
+
+
+# Enablement
+
+The feature will be guarded by a flag `operator-manages-identities`, that can be set in container arguments in Operator’s and Agent’s manifests. The default values for both will be `false`, meaning that Operator doesn’t manage CIDs, while Agents do.
+
+
+
+# Migration
+
+Any situation when neither Operator nor Agents are managing CIDs is effectively a downtime for assigning new Security Identities. Either of these components must be available and healthy in order to:
+- Deploy workloads with a new set of identity-relevant labels
+- Update existing workloads with different identity-relevant labels
+- Modify the identity-relevant labels filter
+- Modify namespace labels and enforce policies for the newly changed labels
+
+
+## Upgrade
+
+**From Agents managing CIDs to Operator managing CIDs”**
+
+Safe upgrade, without downtime, requires simultaneous CID management compatibility. The CID controller in Operator is designed to support simultaneous CID management and won’t interfere with Agents.
```

Comment:
Let's update this based on the latest discussion on upgrade. For instance, "safe upgrade, without downtime" is vague and doesn't directly reference what kind of downtime / impact the upgrade may have.

Reply:
I updated the CFP based on our discussion. I described the type of disruptions and which ones can occur in which situations. Both for upgrade and downgrade.